server/src/main/kotlin/org/jetbrains/bsp/bazel/server/bep/BepServer.kt

```diff
@@ -284,7 +284,8 @@ class BepServer(
   }
 
   private fun consumeCompletedEvent(event: BuildEventStreamProtos.BuildEvent) {
-    val label = event.id.targetCompleted.label
+    val eventLabel = event.id.targetCompleted.label
+    val label = if (target != null && ("@$eventLabel" == target.uri || "@@$eventLabel" == target.uri) ) target.uri else eventLabel
```

Comment:
can you explain the logic behind this?

Reply:
It looks like sometimes the label is without `@` or `@@` so we don't identify the target properly. And for the on progress updates we sue the actual build target label

Reply:
hmm, do you have any idea why it sometimes contains `@` and sometimes not?

Reply:
I think it might be a bug in Bazel, I think this is why I had to do similar fix before. I can dig in a bit more later.

Reply:
Ach, I remember now, the events never contain `@` this is the exact same reason we had to do the workaround in https://github.com/JetBrains/bazel-bsp/pull/534/files

So here this works around that fact, but if we set up the BEP server to gather info about build targets within certain path (//... etc.) then this will not break. So this only fixes the diagnotics.

The problem happens because in progress notifications we don't get the target label, so we use whatever BEP was set up with.

Reply:
Can you put that as a comment in the code?

Reply:
Done, this still isn't perfect, since if we retry a compilation on a target, which was already compiler with warnings they will be cleaned as the build is cached and no new events are produced. But this is a different issue not sure how to fix it properly yet.

Reply:
Though I think it's still an improvement as before this PR warnings would just disappear after any changes, @agluszak does it make sense to merge?